js/src/tooltip.js

```diff
@@ -761,6 +768,16 @@ class Tooltip {
     this.config.animation = initConfigAnimation
   }
 
+  _getCustomClass() {
+    const customClass = Manipulator.getDataAttribute(this.element, 'custom-class')
+
+    if (customClass) {
+      return customClass
+    }
+
+    return typeof this.config.customClass === 'function' ? this.config.customClass() : this.config.customClass
+  }
+
```

Comment:
I have done this as per #31834 though but IMO, `customClass` in the config should have more priority than in data attributes 🤔